### ISSUE TITLE
Type instantiation is excessively deep and possibly infinite

### ISSUE THREAD

**Issue by maxdzin** [intlify/eslint-plugin-vue-i18n#485]
This plugin attempts to be a great help with making the code clear, so first of all, I want to thank you for creating it.

There's a problem now when using this plugin. Starting from version `3.0.0-next.2` and up to the current `3.0.0-next.7` it caused `typecheck` command to fail with an error:
`error TS2589: Type instantiation is excessively deep and possibly infinite.`

Please, note that before `next` version it was working properly and didn't inflict such a thing.

**Tell us about your environment**

- **ESLint version:** 8.57.0
- **eslint-plugin-vue version:** -
- **eslint-plugin-vue-i18n version:** 3.0.0-next.7
- **Node version:** 20.11.1

**Please show your full configuration:**

I prepared the reproduction repository with a couple of branches with different configurations (see below).

**What did you do?**

Just try to use some translation in a vue file template:

```vue
<template>
  <div>
    <span>{{ $t('app.intro.title') }}</span>
  </div>
</template>
```

**What did you expect to happen?**
Everything works well and running `typecheck` didn't show the mentioned error.

**What actually happened?**

When launching `pnpm typecheck` it fails with an error:
```bash
error TS2589: Type instantiation is excessively deep and possibly infinite.
```

Please, note that even without specifying the plugin in eslint configuration or even without eslint setup in project at all, it caused `typecheck` to fail - just by having the plugin installed as a dev dependency (see the `initial` branch of reproduction repository for that case).

**Repository to reproduce this issue**

As mentioned, I created a repository with a couple of branches for several configurations:
* attempt to use flat config - https://github.com/maxdzin/nuxt-i18n-test
* using eslintrc - https://github.com/maxdzin/nuxt-i18n-test/tree/eslint-rc
* initial minimal set to reproduce - https://github.com/maxdzin/nuxt-i18n-test/tree/initial

Launching `pnpm typecheck` in any of the branches above fails with an error.

**Comment by ferferga** [comment]
This is an issue with your setup. Check https://github.com/intlify/bundle-tools/pull/272 and https://github.com/intlify/vue-i18n-next/issues/1119#issuecomment-1351270253

This is not an ESLint error, since TypeScript doesn't have anything to do with ESLint.

**Comment by maxdzin** [comment]
> This is not an ESLint error, since TypeScript doesn't have anything to do with ESLint.
There's something wrong when using this plugin with [@nuxtjs/i18n module](https://github.com/nuxt-modules/i18n), but not only. As I stated above, the problem appears even with just simply installing this plugin as a dependency.
I'm not yet sure what is wrong there exactly, since I haven't time to check that deeply yet.

**Comment by verynong** [comment]
I'm currently experiencing this problem
TS2589 Type instantiation is excessively deep and possibly infinite.
![image](https://github.com/user-attachments/assets/dfd938d8-9ab5-4c29-a7e6-26cc2c949de4)
![image](https://github.com/user-attachments/assets/02101b33-63f0-4976-9b24-f5e983902151)

**Comment by oliver139** [comment]
Same with `te` too
having same TS error, but the code can still run.
I am now simply ignoring it with `// @ts-expect-error`
